crates/hir-def/src/body/lower.rs

```diff
@@ -1145,10 +1206,44 @@ impl ExprCollector<'_> {
                     statements.push(Statement::Expr { expr, has_semi });
                 }
             }
-            ast::Stmt::Item(_item) => statements.push(Statement::Item),
+            ast::Stmt::Item(ast::Item::MacroDef(macro_)) => {
+                let Some(name) = macro_.name() else {
+                    statements.push(Statement::Item(Item::Other));
+                    return;
+                };
+                let name = name.as_name();
+                let macro_id = self.def_map.modules[DefMap::ROOT]
+                    .scope
+                    .get(&name)
+                    .take_macros()
+                    .expect("def map should have macro definition, but it doesn't");
```

Comment:
```suggestion
                    .expect("def map should have macro definition, but it doesn't");
```
Might be better to skip with a `never!` instead, this feels like we could easily write a bug that reaches this